[DOCS] Fixes links in Developer docs (#81843)
elastic/kibana: docs/developer/contributing/development-functional-tests.asciidoc

[discrete]
=== Running functional tests

> The `FunctionalTestRunner` is very bare bones and gets most of its functionality from its config file, located at link:{kib-repo}/tree/{branch}/test/functional/config.js[test/functional/config.js]. If you’re writing a plugin outside the {kib} repo, you will have your own config file.

Reviewer: Doesn't appear to cause any issues to the links, but it is resulting in two slashes. I assume `kib-repo` includes a trailing slash?

https://github.com/elastic/kibana//tree/7.10/test/functional/config.js

```suggestion
The `FunctionalTestRunner` is very bare bones and gets most of its functionality from its config file, located at link:{kib-repo}tree/{branch}/test/functional/config.js[test/functional/config.js]. If you’re writing a plugin outside the {kib} repo, you will have your own config file.
```

Reviewer: There are still quite a few instances with a slash before `tree` if you want to remove them. I had only done a commit suggestion on this one instance.

node scripts/functional_test_runner
----------

> ** If you are running x-pack functional tests, start server and runner from the link:{kib-repo}/tree/{branch}/x-pack[x-pack] folder:

Reviewer: Same as above, the resulting link is currently https://github.com/elastic/kibana//tree/7.10/x-pack

```suggestion
** If you are running x-pack functional tests, start server and runner from the link:{kib-repo}tree/{branch}/x-pack[x-pack] folder:
```

export TEST_THROTTLE_NETWORK=1
----------

> ** When running against a Cloud deployment, some tests are not applicable. To skip tests that do not apply, use --exclude-tag.  An example shell file can be found at link:{kib-repo}/tree/{branch}/test/scripts/jenkins_cloud.sh[test/scripts/jenkins_cloud.sh]

Reviewer 2: What is the significance of the `link:` prefix here?

Author: @spalger that was my way of signifying to myself that the link was broken. I think `link:` tells Asciidoc that the link is external to the documentation.

> ** When running against a Cloud deployment, some tests are not applicable. To skip tests that do not apply, use --exclude-tag.  An example shell file can be found at link:{kib-repo}/tree/{branch}/test/scripts/jenkins_cloud.sh[test/scripts/jenkins_cloud.sh]

Reviewer: Same as above, the resulting link is currently https://github.com/elastic/kibana//tree/7.10/test/scripts/jenkins_cloud.sh

```suggestion
** When running against a Cloud deployment, some tests are not applicable. To skip tests that do not apply, use --exclude-tag.  An example shell file can be found at link:{kib-repo}/tree/{branch}test/scripts/jenkins_cloud.sh[test/scripts/jenkins_cloud.sh]
```

We use https://www.w3.org/TR/webdriver1/[WebDriver Protocol] to run tests in both Chrome and Firefox with the help of https://sites.google.com/a/chromium.org/chromedriver/[chromedriver] and https://firefox-source-docs.mozilla.org/testing/geckodriver/[geckodriver]. When the `FunctionalTestRunner` launches, remote service creates a new webdriver session, which starts the driver and a stripped-down browser instance. We use `browser` service and `webElementWrapper` class to wrap up https://seleniumhq.github.io/selenium/docs/api/javascript/module/selenium-webdriver/[Webdriver API].

> The `FunctionalTestRunner` automatically transpiles functional tests using babel, so that tests can use the same ECMAScript features that {kib} source code uses. Refer to {kib-repo}/tree/{branch}/style_guides/js_style_guide.md[style_guides/js_style_guide.md].

Reviewer 3: Can link to https://github.com/elastic/kibana/blob/7.10/STYLEGUIDE.md#typescriptjavascript

```suggestion
The `FunctionalTestRunner` automatically transpiles functional tests using babel, so that tests can use the same ECMAScript features that {kib} source code uses. Refer to {kib-repo}/tree/{branch}/STYLEGUIDE.md#typescriptjavascript[STYLEGUIDE.md].
```
